### PR TITLE
Custom Airport Data

### LINKED ISSUE #1225: Allow option for custom airport data
## Describe the feature you'd like to see
Currently, the fraction computing formula is based on ratio size of airport, which has two categories which is medium and large. 

@danielelerede-oet and @yerbol-akhmetov together with me are currently working on a US model in which we have found passenger data for each airport to have a positive correlation of about 92% to consumption. We will like to use this information on passenger data to split aviation demand into each airports. we have our data to in the same structure as the same csv result from prepare_airports.py.

If there is also any information regarding passenger data for all airports in the world, we will really appreciate the information.

The idea is to have a flag in config that can allows usage of custom airports data.

@ekatef @davide-f @hazemakhalek @energyLS @Eddy-JV

### PR DESCRIPTION
**Pull request by GbotemiB**
# Closes #1225 .

## Changes proposed in this Pull Request
This PR aims to address the use of custom data for airports data. This PR is a follow-up from #1225 

## Checklist

- [x] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `doc/requirements.txt`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by ekatef** [comment]
Super @GbotemiB! Thank you for the implementation. On the first glance, looks perfectly reasonable. Would be great to discuss the details tomorrow, also with the experts on sector-coupled modeling

File: data/custom/airports.csv [hunk 1]
```diff
@@ -0,0 +1,2 @@
+,ident,type,airport_size_nr,name,y,x,elevation_ft,continent,country,iso_region,municipality,scheduled_service,iata_code,length_ft,width_ft,surface,lighted,closed,Total_airport_size_nr,fraction
+0,5A8,medium_airport,1,Aleknagik / New Airport,59.2826004028,-158.617996216,66.0,,US,US-AK,Aleknagik,yes,WKK,2040.0,80.0,GRVL-DIRT-F,0.0,0.0,633,0.001579778830963665
```
Comment (yerbol-akhmetov) [inline data/custom/airports.csv:2]: Thanks, @GbotemiB. Was the second line given as a example?

Comment (GbotemiB) [reply data/custom/airports.csv:2]: Yeah you are right, the second line is supposed to be as an example.

**Comment by Eddy-JV** [comment]
Good work @GbotemiB . Thank you for implementing that. Sadly,I could not find passenger information for all airports globally. But for now, this is a good solution for  countries who have such more realistic data for consumption.

I will merge now.

**Comment by ekatef** [comment]
@GbotemiB @Eddy-JV @yerbol-akhmetov perfect 😄 Great work!! 🎉

**Pull request merged**